### PR TITLE
[JENKINS-49839] correct annotation check

### PR DESCRIPTION
**Pull request by Wadeck**
After having some tests that were failing due to some bad annotation placement and further check on the global configuration, lots of implementation were not checking correctly for nullability. Especially  for the `get` method.
- also correct a bad annotation in `Run` (two usages, one passing null, asking for `@Nonnull` but checking for nullability inside)
- replace lots of `Jenkins.getInstance()` by the new `Jenkins.get()`

See [JENKINS-49839](https://issues.jenkins-ci.org/browse/JENKINS-49839).

### Proposed changelog entries

* Developer: better annotation usage in `GlobalConfiguration`'s implementations

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)

### Desired reviewers

@reviewbybees

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by Wadeck** [comment]
> I would like to review it next week if you do not mind

With pleasure 👍

**Review by oleg-nenashev — APPROVED** [review]

File: core/src/main/java/jenkins/model/GlobalConfigurationCategory.java [hunk 1]
```diff
@@ -41,8 +41,12 @@
         return ExtensionList.lookup(GlobalConfigurationCategory.class);
     }
 
-    public static <T extends GlobalConfigurationCategory> T get(Class<T> type) {
-        return all().get(type);
+    public static @Nonnull <T extends GlobalConfigurationCategory> T get(Class<T> type) {
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/GlobalConfigurationCategory.java]: Add Javadoc?
BTW, I didn't find ant usage of the method which would actually check for null.

A conservative approach would be to make the method `@CheckForNull`, but I am fine with the current impl in this particular case

**Comment by oleg-nenashev** [comment]
@reviewbybees done, but I am concerned about some bits in API. Let's give other reviewers some time to comment

**Comment by daniel-beck** [comment]
@oleg-nenashev  Ping

**Comment by oleg-nenashev** [comment]
@daniel-beck My concern is above: https://github.com/jenkinsci/jenkins/pull/3321#pullrequestreview-112219347

**Pull request merged**